ezyhttp-server-graphql/src/main/java/com/tvd12/ezyhttp/server/graphql/controller/GraphQLController.java

```diff
@@ -0,0 +1,168 @@
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tvd12.ezyfox.builder.EzyBuilder;
+import com.tvd12.ezyhttp.core.exception.HttpNotFoundException;
+import com.tvd12.ezyhttp.server.core.annotation.Controller;
+import com.tvd12.ezyhttp.server.core.annotation.DoGet;
+import com.tvd12.ezyhttp.server.core.annotation.DoPost;
+import com.tvd12.ezyhttp.server.core.annotation.RequestBody;
+import com.tvd12.ezyhttp.server.core.annotation.RequestParam;
+import com.tvd12.ezyhttp.server.graphql.GraphQLDataFetcher;
+import com.tvd12.ezyhttp.server.graphql.GraphQLDataFetcherManager;
+import com.tvd12.ezyhttp.server.graphql.GraphQLField;
+import com.tvd12.ezyhttp.server.graphql.GraphQLRequest;
+import com.tvd12.ezyhttp.server.graphql.GraphQLSchema;
+import com.tvd12.ezyhttp.server.graphql.GraphQLSchemaParser;
+
+@Controller
+public class GraphQLController {
+
+	private final ObjectMapper objectMapper;
+	private final GraphQLSchemaParser schemaParser;
+	private final GraphQLDataFetcherManager dataFetcherManager;
+	
+	public GraphQLController(Builder builder) {
+		this.objectMapper = builder.objectMapper;
+		this.schemaParser = builder.schemaParser;
+		this.dataFetcherManager = builder.dataFetcherManager;
+	}
+	
+	/**
+	 * Follow by this suggestion: https://graphql.org/learn/serving-over-http/ 
+	 * 
+	 * @param operationName be used to control which one should be executed.
+	 * @param query GraphQL query
+	 * @param variables a JSON-encoded string like <code>{ "myVariable": "someValue", ... }</cocde>
+	 * @return the result
+	 * @throws Exception when have any error
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	@DoGet("/graphql")
+	public Object doGet(
+			@RequestParam("operationName") String operationName,
+			@RequestParam("query") String query,
+			@RequestParam("variables") String variables
+	) throws Exception {
+		GraphQLDataFetcher dataFetcher = dataFetcherManager.getDataFetcher(operationName);
+		if(dataFetcher == null) {
+			throw new HttpNotFoundException("not found data fetcher with operationName: " + operationName);
+		}
+		Class<?> argumentType = dataFetcher.getArgumentType();
+		Object argument = null;
+		if(argumentType != null) {
+			argument = objectMapper.readValue(variables, argumentType);
+		}
+		else {
+			argument = objectMapper.readValue(variables, Map.class);
+		}
+		Object data = dataFetcher.getData(argument);
+		return mapToResponse(data, query);
+	}
+	
+	/**
+	 * Follow by this suggestion: https://graphql.org/learn/serving-over-http/
+	 * 
+	 * @param request the request body
+	 * @return the result
+	 * @throws Exception when have any error
+	 */
+	@DoPost
+	private Object doPost(@RequestBody GraphQLRequest request) throws Exception {
+		return doGet(
+				request.getOperationName(), 
+				request.getQuery(),
+				request.getVariables()
+		);
+	}
```

Comment:
Chỗ này là @DoPost("/graphql") phải không anh? Mà em sửa lại vẫn không gửi được POST. Nó báo `method POST not allowed`